cmd/openfga/main_test.go

```diff
@@ -649,6 +706,117 @@ func GRPCExpandTest(t *testing.T, tester OpenFGATester) {
 
 }
 
+// TestCheckWorkflows are tests that involve workflows that define assertions for
+// Checks against multi-model stores etc..
+func TestCheckWorkflows(t *testing.T) {
+
+	tester, err := newOpenFGATester(t)
+	require.NoError(t, err)
+	defer tester.Cleanup()
+
+	conn := connect(t, tester)
+	defer conn.Close()
+
+	client := openfgapb.NewOpenFGAServiceClient(conn)
+
+	/*
+	 * TypedWildcardsFromOtherModelsIgnored ensures that a typed wildcard introduced
+	 * from a prior model does not impact the Check outcome of a model that should not
+	 * involve it. For example,
+	 *
+	 * type user
+	 * type document
+	 *   relations
+	 *     define viewer: [user:*]
+	 *
+	 * write(document:1#viewer@user:*)
+	 *
+	 * type user
+	 * type document
+	 *	relations
+	 *	  define viewer: [user]
+	 *
+	 * check(document:1#viewer@user:jon) --> {allowed: false}
+	 */
+	t.Run("TypedWildcardsFromOtherModelsIgnored", func(t *testing.T) {
```

Comment:
This test tests the workflow described in the PR description which involves two different models in the same store and a typed wildcard tuple that is interpreted differently based on the model schema_version involved in the Check query.